clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -18,24 +18,43 @@ package com.netflix.spinnaker.clouddriver.openstack.client
 
 import com.netflix.spinnaker.clouddriver.openstack.deploy.description.securitygroup.OpenstackSecurityGroupDescription
 import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackOperationException
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackProviderException
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
 import org.apache.commons.lang.StringUtils
 import org.openstack4j.api.Builders
 import org.openstack4j.api.OSClient
-import org.openstack4j.api.compute.ComputeSecurityGroupService
 import org.openstack4j.model.common.ActionResponse
 import org.openstack4j.model.compute.IPProtocol
 import org.openstack4j.model.compute.RebootType
 import org.openstack4j.model.compute.SecGroupExtension
+import org.openstack4j.model.compute.Server
 import org.openstack4j.model.heat.Stack
+import org.openstack4j.model.network.ext.LbPool
+import org.openstack4j.model.network.ext.Member
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
 
 /**
  * Provides access to the Openstack API.
  *
+ * TODO use OpenstackProviderException here instead of OpenstackOperationException.
+ * Use of OpenstackOperationException belongs in Operation classes.
+ *
+ * TODO handleRequest should be refactored to remove the operation parameter, as client calls
+ * made here do not necessarily pertain to operations.
+ *
+ * TODO region support will need to be added to all client calls not already using regions
+ *
  * TODO tokens will need to be regenerated if they are expired.
  */
 abstract class OpenstackClientProvider {
 
+  final int minPort = 1
```

Comment:
By convention, constants should be named in all caps, but I see this isn't necessarily observed in other providers in this code base...